pkg/Cpanel/Security/Advisor/Assessors/Trojans.pm

```diff
@@ -73,7 +92,521 @@ sub _check_for_libkeyutils {
+    my @found_jynx2_files = grep { -e } map { "$dir/$_" } qw( 3.so 4.so );
+    if ( scalar @found_jynx2_files > 0 ) {
+        $self->add_bad_advice(
+            'text'       => [ "Jynx 2 rootkit check: Evidence of the Jynx 2 rootkit was found." ],
+            'suggestion' => [
+                'Check the following to determine if this server is compromised "[output,url,_1,Determine your Systems Status,_2,_3]"',
+                'https://documentation.cpanel.net/display/CKB/Determine+Your+System%27s+Status',
+                'target',
+                '_blank'
+            ],
+        );
+    }
+}
+
+sub _check_for_cdorked_A {
+    my ($self) = @_;
+    my $apache_bin   = '/usr/local/apache/bin/httpd';
+    my $max_bin_size = 10_485_760;                      # avoid slurping too much mem
+    return if ( !-f $apache_bin );
+    return if ( ( stat($apache_bin) )[7] > $max_bin_size );
+
+    my $has_cdorked = 0;
+    my $signature;
+    my @apache_bins = ();
+    push @apache_bins, $apache_bin;
+    my @PROCESS_LIST;
+    for my $process (@PROCESS_LIST) {
+        if ( $process =~ m{ \A root \s+ (\d+) [^\d]+ $apache_bin }xmsa ) {
+            my $pid          = $1;
+            my $proc_pid_exe = "/proc/" . $pid . "/exe";
+            if ( -l $proc_pid_exe && readlink($proc_pid_exe) =~ m{ \(deleted\) }xms ) {
+                next if ( ( stat($proc_pid_exe) )[7] > $max_bin_size );
+                push @apache_bins, $proc_pid_exe;
+            }
+        }
+    }
+
+    for my $check_bin (@apache_bins) {
+        my $httpd;
+        if ( open my $fh, '<', $check_bin ) {
+            local $/;
+            $httpd = <$fh>;
+            close $fh;
+        }
+
+        next if !$httpd;
+
+        if ( $httpd =~ /(open_tty|hangout|ptsname|Qkkbal)/ ) {
+            $signature   = $check_bin . ": \"" . $1 . "\"";
+            $has_cdorked = 1;
+            last;
+        }
+    }
+    if ( $has_cdorked == 1 ) {
+        $self->add_bad_advice(
+            'text'       => [ "CDORKED rootkit check: Evidence of the CDORKED A rootkit was found." ],
+            'suggestion' => [
+                'Check the following for more information "[output,url,_1,We Live Security More Information,_2,_3]"',
+                'http://www.welivesecurity.com/wp-content/uploads/2014/03/operation_windigo.pdf',
+                'target',
+                '_blank'
+            ],
+        );
+    }
+}
+
+sub _check_for_cdorked_B {
+    my ($self) = @_;
+    my $has_cdorked_b = 0;
+    my @files = ( '/usr/sbin/arpd ', '/usr/sbin/tunelp ', '/usr/bin/s2p ' );
+    my $cdorked_files;
+
+    for my $file (@files) {
+        if ( -e $file ) {
+            $has_cdorked_b = 1;
+            $cdorked_files .= "[$file] ";
+        }
+    }
+
+    if ( $has_cdorked_b == 1 ) {
+        $self->add_bad_advice(
+            'text'       => [ "CDORKED rootkit check: The following file(s) [_1] was/were found, indicating the possibility of the CDORKED B rootkit.", $cdorked_files ],
```

Comment:
I recommend the following string for better readability:

```perl
[ "CDORKED rootkit check: The following [numerate,_1,file was,files were] found, indicating the possibility of the CDORKED B rootkit: [list_and,_1]", \@cdorked_files ]
```

Where `my @cdorked_files = grep { -e } @files;`.